Core/src/IfcGeometryConverter/SolidModelConverter.h

```diff
@@ -1056,195 +1030,158 @@ namespace OpenInfraPlatform
-					//{
-					//	profile_loop_2d.push_back( carve::geom::VECTOR( profile_loop[j].x, profile_loop[j].y ) );
-					//}
-					profile_coords_2d.push_back(profile_loop);
-				}
-
-				std::vector<carve::geom::vector<2> > merged;
+				std::vector<std::vector<carve::geom::vector<2>>> profileCoords2D = profileCoords;
+			
+				std::vector<carve::geom::vector<2>> merged;
 				std::vector<carve::triangulate::tri_idx> triangulated;
-				try
+
+				try //TODO> Decide, if it is necessary here to have try ... catch (...).
 				{
-					std::vector<std::pair<size_t, size_t> > result = carve::triangulate::incorporateHolesIntoPolygon(profile_coords_2d);	// first is loop index, second is vertex index in loop
+					std::vector<std::pair<size_t, size_t>> result = carve::triangulate::incorporateHolesIntoPolygon(profileCoords2D);	// first is loop index, second is vertex index in loop
 					merged.resize(result.size());
 					for (size_t i = 0; i < result.size(); ++i)
 					{
-						int loop_number = result[i].first;
-						int index_in_loop = result[i].second;
+						int loopNumber = result[i].first;
+						int indexInLoop = result[i].second;
 
-						if (loop_number >= profile_coords_2d.size())
+						if (loopNumber >= profileCoords2D.size())
 						{
-							std::cout << ": loop_number >= face_loops_projected.size()" << std::endl;
+							std::cout << ": loopNumber >= face_loops_projected.size()" << std::endl;
 							continue;
 						}
 
-						std::vector<carve::geom2d::P2>& loop_projected = profile_coords_2d[loop_number];
+						std::vector<carve::geom2d::P2>& loopProjected = profileCoords2D[loopNumber];
 
-						carve::geom2d::P2& point_projected = loop_projected[index_in_loop];
-						merged.push_back(point_projected);
+						carve::geom2d::P2& pointProjected = loopProjected[indexInLoop];
+						merged.push_back(pointProjected);
 					}
 					carve::triangulate::triangulate(merged, triangulated);
 					carve::triangulate::improve(merged, triangulated);
 				}
 				catch (...)
 				{
-					/*for( size_t i = 0; i < profile_coords_2d.size(); ++i )
-					{
-					for ( size_t j = 0; j < profile_coords_2d[i].size(); ++j)
-					{
-					int loop_number = i;
-					int index_in_loop = j;
-
-					carve::geom2d::P2& loop_point = profile_coords_2d[loop_number][index_in_loop];
-					merged.push_back( loop_point );
-
-					if( loop_number >= profile_coords_2d.size() )
-					{
-					std::cout << __FUNC__ << ": loop_number >= face_loops_projected.size()" << std::endl;
-					continue;
-					}
-
-					std::vector<carve::geom2d::P2>& loop_projected = profile_coords_2d[loop_number];
-
-					carve::geom2d::P2& point_projected = loop_projected[index_in_loop];
-					merged.push_back( point_projected );
-					}
-					}
-					carve::triangulate::triangulate(merged, triangulated);
-					carve::triangulate::improve(merged, triangulated);*/
-
-
 					BLUE_LOG(error) << "carve::triangulate::incorporateHolesIntoPolygon failed ";
 					return;

```

Comment:
Either/or:

- Consider introducing a new exception
- throw a geometry exception

Reply:
Right now I've decided to stay with InconsistentGeometryException, but I will try to introduce a new exception after I am done with this. Resolved: 4c40ba7